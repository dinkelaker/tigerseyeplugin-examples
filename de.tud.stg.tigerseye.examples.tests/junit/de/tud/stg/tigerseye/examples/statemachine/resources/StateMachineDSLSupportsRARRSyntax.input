state(name:"reseted",type:"start") {
  entry "resetTimer"
  transitions {
  	"start" → "running" 
  	"switchOff" → "off"
  }
};

state(name:"running") {
  entry "startTimer"
  perform "loopRunning" //do is a reserved keyword in Groovy
  transitions {
  "split" → "paused"
  "stop" → "stopped"
  }
};

state(name:"paused") {
  entry "pauseTimer"
  transitions {
  	"unsplit" → "running"
  	"stop" → "stopped"
  }
};

state(name:"stopped") {
  entry "stopTimer"
  transitions {
  "reset" → "stopped"
  "switchOff" → "off"
  }
};

state(name:"off") {
  exit "switchOff"
  transitions {
	→ "end"
  }
};
