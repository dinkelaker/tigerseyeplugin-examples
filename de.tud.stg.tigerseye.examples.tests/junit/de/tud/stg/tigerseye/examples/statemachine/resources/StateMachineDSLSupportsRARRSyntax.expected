
		state(name:"reseted",type:"start") {
		  entry "resetTimer"
		  transitions {
			when_event_enter("start","running")
			when_event_enter("switchOff","off")
		  }
		};
	
		state(name:"running") {
		  entry "startTimer"
		  perform "loopRunning" //do is a reserved keyword in Groovy
		  transitions {
			when_event_enter("split","paused")
			when_event_enter("stop","stopped")
		  }
		};
	
		state(name:"paused") {
		  entry "pauseTimer"
		  transitions {
			when_event_enter("unsplit","running")
			when_event_enter("stop","stopped")
		  }
		};
	
		state(name:"stopped") {
		  entry "stopTimer"
		  transitions {
			when_event_enter("reset","stopped")
			when_event_enter("switchOff","off")
		  }
		};
	
		state(name:"off") {
		  exit "switchOff"
		  transitions {
			when_enter("end")
		  }
		};